pkg/controller/directvolumemigration/rsync.go

```diff
@@ -742,6 +745,42 @@ func (t *Task) getPVCNamespaceMap() map[string][]pvcMapElement {
 	return nsMap
 }
 
+func (t *Task) getDestPVCNamespaceMap() map[string][]pvcMapElement {
+	nsMap := map[string][]pvcMapElement{}
+	for _, pvc := range t.Owner.Spec.PersistentVolumeClaims {
+		ns := pvc.Namespace
+		if pvc.TargetNamespace != "" {
+			ns = pvc.TargetNamespace
+		}
+		if vols, exists := nsMap[ns]; exists {
+			vols = append(vols, pvcMapElement{Name: pvc.Name, Verify: pvc.Verify})
+			nsMap[ns] = vols
+		} else {
+			nsMap[ns] = []pvcMapElement{{Name: pvc.Name, Verify: pvc.Verify}}
+		}
+	}
+	return nsMap
+}
+
+func (t *Task) getBothPVCNamespaceMap() map[string][]pvcMapElement {
+	nsMap := map[string][]pvcMapElement{}
+	for _, pvc := range t.Owner.Spec.PersistentVolumeClaims {
+		srcNs := pvc.Namespace
+		destNs := srcNs
+		if pvc.TargetNamespace != "" {
+			destNs = pvc.TargetNamespace
+		}
+		bothNs := srcNs + ":" + destNs
```

Comment:
Don't feel strongly about it, but something to consider, the only difference between the two functions `get<>PVCMap()` is the above 1 line. How about we keep only the `getBothPVCNamespaceMap()` function which will contain mapping by default, and a set of helper functions to simply split the key in the map, something like `getSourceNs()` & `getDestNs()`. The helper functions will abstract the mapping logic. So anyone else who isn't familar with the change can simply re-use the helpers.

Reply:
Also, @djwhatle's suggestion about length of array can go into the helper.

Reply:
@pranavgaikwad So if I"m understanding correctly, you're saying that of the 3 get...PVCMap functions, we only retain the getBoth() one and that the individual callers will call getSourceNs() or getDestNS() on each key element returned in their respective loops? If that's correct, then yes, that makes sense and I can make that change.